File: src/runtime/apis/clipboard.ts
import vokexCall from "./vokexCall";
/**
 * Clipboard API 接口
 */
export interface ClipboardAPI {
  /** 读取剪贴板文本 */
  readText: () => Promise<string>;
  /** 写入文本到剪贴板 */
  writeText: (text: string) => Promise<void>;
  /** 清空剪贴板 */
  clear: () => Promise<void>;
}

/**
 * 剪贴板 API
 */
export const clipboard: ClipboardAPI = {
  /**
   * 读取剪贴板文本
   */
  readText: (): Promise<string> => vokexCall('clipboard.readText', {}),

  /**
   * 写入文本到剪贴板
   * @param text 要写入的文本
   */
  writeText: (text: string): Promise<void> => vokexCall('clipboard.writeText', { text }),

  /**
   * 清空剪贴板
   */
  clear: (): Promise<void> => vokexCall('clipboard.clear', {}),
};

File: shell/src/apis/clipboard.rs
use serde_json::{json, Value};

pub fn handle(method: &str, params: &Value) -> Result<Value, String> {
    match method {
        "clipboard.readText" => {
            let mut clipboard = arboard::Clipboard::new()
                .map_err(|e| format!("Failed to open clipboard: {}", e))?;
            let text = clipboard.get_text()
                .unwrap_or_default();
            Ok(json!(text))
        }

        "clipboard.writeText" => {
            let text = params.get("text").and_then(|v| v.as_str())
                .ok_or("Missing 'text' parameter")?;
            let mut clipboard = arboard::Clipboard::new()
                .map_err(|e| format!("Failed to open clipboard: {}", e))?;
            clipboard.set_text(text.to_string())
                .map_err(|e| format!("Failed to write clipboard: {}", e))?;
            Ok(json!(true))
        }

        "clipboard.clear" => {
            let mut clipboard = arboard::Clipboard::new()
                .map_err(|e| format!("Failed to open clipboard: {}", e))?;
            clipboard.clear()
                .map_err(|e| format!("Failed to clear clipboard: {}", e))?;
            Ok(json!(true))
        }

        _ => Err(format!("Unknown method: {}", method)),
    }
}